### PR TITLE
Add unit tests to spring security pac4j

### PR DESCRIPTION
**Pull request by mfarid**
This pull request is sample of unit test cases that we plan to contribute in order to raise the overall coverage to 80%. Code coverage is measured using Intellij Idea. This pull request increases code coverage of library from 0% to 25%.

Metric | Before | After
-------- | --------- | -----
Coverage % | 0% | 25% 
Lines Covered | 0 | 37
Total Lines | 146 | 146

Please let me know if you have any questions.

Mohd Farid
DevFactory - Code Quality Team

### CODE REVIEW

File: src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
+package org.pac4j.springframework.security.web;
```
Comment (leleuj) [inline src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java]: Can you add our regular header?

File: src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by farid on 27/01/16.
+ */
```
Comment (leleuj) [inline src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java]: Can you add a `@since` and `@author` annotations?

**Comment by leleuj** [comment]
Great! I have a few small comments to address before the merge.

File: src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java (hunk 1 above)
Comment (mfarid) [inline src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java]: Yes, definitely. I all add the same.

File: src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java (hunk 2 above)
Comment (mfarid) [inline src/test/java/org/pac4j/springframework/security/web/ClientAuthenticationFilterTest.java]: Yes, definitely. I will add the same as well.

**Comment by mfarid** [comment]
Hi @leleuj  
I have made the suggested changes. Please review it again.

**Comment by leleuj** [comment]
Perfect. Let's merge.

**Pull request merged**